src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.SkeletonReferenceCache.cs

```diff
@@ -248,15 +252,20 @@ private sealed class SkeletonReferenceSet
             private readonly DeferredDocumentationProvider _documentationProvider;
 
             /// <summary>
-            /// Use WeakReference so we don't keep MetadataReference's alive if they are not being consumed. 
-            /// Note: if the weak-reference is actually <see langword="null"/> (not that it points to null),
-            /// that means we know we were unable to generate a reference for those properties, and future
-            /// calls can early exit.
+            /// The actual assembly metadata produced from the data pointed to in <see cref="_storage"/>.
+            /// </summary>
+            private readonly AsyncLazy<AssemblyMetadata?> _metadata;
+
+            /// <summary>
+            /// Mapping from different values of <see cref="MetadataReferenceProperties"/> to the actual <see
+            /// cref="MetadataReference"/> produced for it.  Used so that if the same compilation is referenced from
+            /// different projects with/without the same properties that the right references are handed out (and shared
+            /// if possible).
             /// </summary>
             /// <remarks>
             /// This instance should be locked when being read/written.
             /// </remarks>
-            private readonly Dictionary<MetadataReferenceProperties, WeakReference<MetadataReference>?> _metadataReferences = new();
+            private readonly Dictionary<MetadataReferenceProperties, AsyncLazy<MetadataReference?>> _metadataReferences = new();
```

Comment:
same as before, but no longer weak.  As per the OP of this PR, having htis be weak seems like an antipattern.

Reply:
hmmm.. if weak reference is going away, I think that means the primaryWorkspace in OOP is having problem (not stay alive) or forking temporaryWorkspace forked from primaryWorkspace is not getting skeleton assembly as expected and re-creating them. or property instance is not reused properly? 

but I guess with 64bits, holding some skeleton assemblies in memory a bit longer won't matter much now.